mod-source-record-manager-server/src/main/java/org/folio/services/JobExecutionServiceImpl.java

```diff
@@ -245,15 +248,21 @@ public Future<Boolean> completeJobExecutionWithError(String jobExecutionId, Okap
     return jobExecutionDao.getJobExecutionById(jobExecutionId, params.getTenantId())
       .map(optionalJobExecution -> optionalJobExecution
         .orElseThrow(() -> new NotFoundException(format("JobExecution with id '%s' was not found", jobExecutionId))))
-      .map(this::verifyJobExecution)
-      .map(this::modifyJobExecutionToCompleteWithCancelledStatus)
-      .compose(jobExec -> updateJobExecutionWithSnapshotStatus(jobExec, params))
-      .compose(jobExec -> deleteRecordsFromSRSIfNecessary(jobExec, params))
+      .compose(jobExecution -> {
+        try {
+          verifyJobExecution(jobExecution);
```

Comment:
does `verifyJobExecution` return a future or any object?

Reply:
Hi @okolawole-ebsco ! It returns JobExecution object, without Future. I want to add that I will change this chain via "recover()"-method as @Abdulkhakimov said. Will refactor this.

Reply:
@okolawole-ebsco  I changed this approach via "recover"-method. Could you please re-review ?